### ISSUE TITLE
make-dist: Allow excluding files

### ISSUE THREAD

**Issue by fingolfin** [fingolfin/ReleaseTools#3]
Sometimes, a few files in a repository should not be included in the release .tar.gz -- e.g. a `dev/` subdirectory. Add an `--exclude=DIR1,DIR2,FILE,...` option to `make-dist`. Make sure it supports globs, so that one say `--exclude=foo.*`.

Also, if we add a way to store permanent settings for the release process (see issue #2), it should also have an `exclude` sections.

In both cases, it might make sense to also add a symmetric `include` counterpart. This makes it possible to `--exclude=foo.* --include=foo.pdf` which would exclude all `foo.*` files except for `foo.pdf`. Or one could `--exclude=* --include=a,b,c` in order to only put files a, b, c into the tarball.

**Comment by fingolfin** [comment]
An alternative would be to add this information to `PackageInfo.`

**Comment by alex-konovalov** [comment]
@fingolfin wouldn't it be strange to mention in the `PackageInfo` (which is a part of the release) files that are not included in the release? 

For my repositories, I am using https://github.com/gap-packages/example/blob/master/patternscolor.txt - happy to tell more details on how this works.

**Comment by fingolfin** [comment]
There are a lot of strange things in `PackageInfo.g` *shrug*, I merely said it's another option.

Anyway, thank you for your offer, but I don't think that providing a file including regex to select files is something that needs details (esp. if one used .cvsignore, .hgignore, .gitignore, ... before)... 

My main motivation to consider this variant is that for a future GAP package managment system, it would be good to define a uniform way for packages to specify such information. The existing `PackageInfo.g` file seems like a natural candidate -- but of course we could also specify a new file, e.g. `.gap-distro-ignore` (to make up a name).

Anyway, this is nothing I consider important right now :-)

**Comment by fingolfin** [comment]
I thought more about this, and I discovered that I can't think of a scenario where I personally would want that feature. As such, I don't think it makes sense if I implemented it, nor does it make sense to keep this issue open

Don't get me wrong: I am not claiming there are no legitimate use cases, just that I can't imagine I would ever have one. And based on experience, it is a bad idea to implement a feature you never use, without any real-world test case, because you'll just get it wrong.

So I'll just wait and see: Perhaps somebody will turn up who explicitly asks for this feature. Then I'll be happty to discuss and possibly implement it, and/or help that person to implement it. But until that happens, I am closing this.
